docker/common/install_ccache.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -ex
+
+# Install ccache from source.
+# Needs 3.4 or later for ccbin support
+pushd /tmp
+git clone https://github.com/ccache/ccache -b v3.4.1
```

Comment:
why can't we have it from binary repositories?  ubuntu has it.

Reply:
do we want some specific version?  why?

Reply:
may I request that I go over the setup once using the recommended practice and then  I will be able to answer questions better. This PR is still marked as WIP. Thanks for cooperation. :)

Reply:
to answer the question: trusty has ccache 3.1 version which doesn't work with nvcc well. The nvcc support was added in version 3.4.3 so we install this way.